### DISCUSSION TITLE
restart bot if error

### DISCUSSION THREAD

**Discussion by yuretzgt** [python-telegram-bot/python-telegram-bot#3324] [Q&A]
I run my bot in docker + django, with restart always instruction

`  
bot:
    restart: always
    build: .
    command: bash -c "python manage.py runscript offers"
    volumes:
      - .:/code
`

Run with application.run_polling()
   
   but if something does wrong bot doesn't restart because it doesn't stop and waits for Ctrl-C keys.

So the questions is: how to do it propertly?

**Comment by Poolitzer** [comment]
Well you need to define to yourself what "something does wrong" would mean in your context and then do a sys.exit() in that case.

I am very sure you can configure a logger to do this for you on every exception if you want that,

**Reply by yuretzgt** [reply]
sys.exit() doesn't works for me: after message 
`2022-10-31 17:14:53,974 - telegram.ext._application - INFO - Application is stopping. This might take a moment.`

nothing changes

i think script still waits for Ctrl-C keys
